pkg/networkmanager/interfaces.js

```diff
@@ -612,31 +628,61 @@
                 delete result[first][second];
         }
 
-        function set_ip(first, addrs_sig, addr_to_nm, routes_sig, route_to_nm, ips_sig, ip_from_text) {
+        function set_ip(first, dns_ip_sig, ip_from_text) {
             set(first, "method", 's', settings[first].method);
             set(first, "ignore-auto-dns", 'b', settings[first].ignore_auto_dns);
             set(first, "ignore-auto-routes", 'b', settings[first].ignore_auto_routes);
             set(first, "addr-gen-mode", 'i', settings[first].addr_gen_mode);
 
-            const addresses = settings[first].addresses;
+            const addresses = settings[first].address_data;
             if (addresses)
-                set(first, "addresses", addrs_sig, addresses.map(addr_to_nm));
+                set(first, "address-data", "aa{sv}", addresses.map(addr => ip_address_to_nm(addr, first)));
+
+            const gateway = settings[first].gateway;
+            if (gateway && addresses.length > 0) {
+                if (!utils.validate_ip(gateway)) {
+                    throw cockpit.format(_("Invalid gateway address: $0"), gateway);
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22330-1fe4fb60-20250903-092011-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)